From 3d89b94e2a820c7aa4a90b10486d530e6601e5fe Mon Sep 17 00:00:00 2001
From: Yongzhi Chen <ychena@apache.org>
Date: Wed, 2 Dec 2015 17:54:58 -0500
Subject: [PATCH 0577/1164] CDH-39722 HIVE-12517: Beeline's use of failed
 connection(s) causes failures and leaks. (Naveen
 Gangam via Yongzhi Chen)

Change-Id: I29f640f0e06df9a43da95992d6614237eccbc876
---
 .../src/java/org/apache/hive/beeline/Commands.java |    1 +
 1 file changed, 1 insertion(+)

diff --git a/beeline/src/java/org/apache/hive/beeline/Commands.java b/beeline/src/java/org/apache/hive/beeline/Commands.java
index 6706335..2ecbbe9 100644
--- a/beeline/src/java/org/apache/hive/beeline/Commands.java
+++ b/beeline/src/java/org/apache/hive/beeline/Commands.java
@@ -1338,6 +1338,7 @@ public boolean connect(Properties props) throws IOException {
       beeLine.setCompletions();
       return true;
     } catch (SQLException sqle) {
+      beeLine.getDatabaseConnections().remove();
       return beeLine.error(sqle);
     } catch (IOException ioe) {
       return beeLine.error(ioe);
-- 
1.7.9.5

